--- Makefile.orig	2010-03-06 22:16:20.000000000 +0100
+++ Makefile	2010-03-06 22:38:29.000000000 +0100
@@ -108,19 +108,19 @@
 linux-amd64-profile:
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=g++ \
-                CFLAGS='-Wall -pipe -fprofile-arcs -fomit-frame-pointer -O3 -march=k8' \
+                CFLAGS='-Wall -pipe -fprofile-arcs -fomit-frame-pointer -O3' \
 		CXFLAGS='' \
-		LDFLAGS='$(LDFLAGS) -lpthread -lnuma -fprofile-arcs -lstdc++' \
-		opt='$(opt) -DINLINE64 -DCPUS=8 -DNUMA -DLIBNUMA' \
+		LDFLAGS='$(LDFLAGS) -lpthread -pthreads -fprofile-arcs -lstdc++' \
+		opt='$(opt) -DINLINE64 -DCPUS=8' \
 		crafty-make
 
 linux-amd64:
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=g++ \
-                CFLAGS='-Wall -pipe -fbranch-probabilities -fomit-frame-pointer -O3 -march=k8' \
+                CFLAGS='-Wall -pipe -fbranch-probabilities -fomit-frame-pointer -O3' \
 		CXFLAGS='' \
-		LDFLAGS='$(LDFLAGS) -lpthread -lnuma -lstdc++' \
-		opt='$(opt) -DINLINE64 -DCPUS=8 -DNUMA -DLIBNUMA' \
+		LDFLAGS='$(LDFLAGS) -lpthread -pthread -lstdc++' \
+		opt='$(opt) -DINLINE64 -DCPUS=8' \
 		crafty-make
 
 linux:
@@ -128,8 +128,8 @@
 		CC=gcc CXX=g++ \
 		CFLAGS='-pg -Wall -pipe' \
 		CXFLAGS='' \
-		LDFLAGS='$(LDFLAGS) -pg -lstdc++ -lpthread' \
-		opt='$(opt) -DTRACE -DINLINE64 -DCPUS=2' \
+		LDFLAGS='$(LDFLAGS) -pg -lstdc++ -pthread -lpthread' \
+		opt='$(opt) -DTRACE -DINLINE32 -DCPUS=2' \
 		crafty-make
 
 linux-profile:
@@ -322,7 +322,7 @@
 crafty.o: *.c *.h
 
 crafty:	$(objects) egtb.o
-	$(CC) $(LDFLAGS) -o crafty $(objects) egtb.o -lm  $(LIBS)
+	$(CXX) $(LDFLAGS) -o crafty $(objects) egtb.o -lm  $(LIBS)
 
 egtb.o: egtb.cpp
 	$(CXX) -c $(CXFLAGS) $(opts) egtb.cpp

